src/main/java/seedu/address/logic/commands/EditCommand.java

```diff
@@ -88,6 +90,42 @@ public CommandResult execute(Model model) throws CommandException {
         return new CommandResult(String.format(MESSAGE_EDIT_PERSON_SUCCESS, Messages.format(editedPerson)));
     }
 
+    /**
+     * Finds the person to edit based on the provided name and/or NRIC.
+     *
+     * @param persons A list of persons to search within.
+     * @return An Optional containing the person to edit, if found, or an empty Optional if not found.
+     * @throws CommandException if there is inconsistency between the provided name and NRIC.
+     */
+    public Optional<Person> findPersonToEdit(List<Person> persons) throws CommandException {
+        if (name != null && nric != null) {
+            // Search for a person by name and NRIC
+            Optional<Person> personByName = persons.stream()
+                    .filter(person -> person.getName().equals(name))
+                    .findFirst();
+            Optional<Person> personByNric = persons.stream()
+                    .filter(person -> person.getNric().equals(nric))
+                    .findFirst();
+            // Check if both Optional instances are not empty, and return the one that represents the same person
+            if (personByName.isPresent() && personByNric.isPresent() && personByName.get() == personByNric.get()) {
+                return personByName;
+            } else {
+                throw new CommandException(MESSAGE_INCONSISTENT_NAME_AND_ID);
+            }
+        }
+        if (name != null) {
+            return persons.stream()
+                    .filter(person -> person.getName().equals(name))
+                    .findFirst();
+        } else if (nric != null) {
+            return persons.stream()
+                    .filter(person -> person.getNric().equals(nric))
+                    .findFirst();
+        } else {
+            return Optional.empty();
+        }
+    }
+
```

Comment:
I had left [a similar comment](https://github.com/AY2324S1-CS2103T-T14-3/tp/pull/72#discussion_r1361546213) in a previous review - leaving a note to flag this method as one more suited for the `FindCommand` or a related utility.

This issue is actually quite interesting - since we are building a dependency to this kind of search utility in multiple `Command` classes like `delete`, `edit` and `find`, should we actually move this method to an Interface?